.github/workflows/build.yml

```diff
@@ -0,0 +1,65 @@
+name: Build VS Code extension
+
+on:
+  push:
+  pull_request:
+  release:
+    types: [published]
+
+permissions:
+  # Write permissions needed for publishing the release
+  contents: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  build_vscode_ext:
+    name: Build VS Code extension
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          cache: "npm"
+
+      - name: Install Javascript dependencies
+        run: npm ci
+
+      - name: Check formatting
+        run: npm run format-check
+
+      # Has to happen before `check-lint`, because we need the generated protobuf types
+      - name: Compile the extension
+        run: npm run compile
+
+      - name: Lint
+        run: npm run check-lint
+
+      - name: Package VS Code extension
+        run: npm run package
+
+      - name: Upload Workflow Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: vscode-bazel-prerelease.vsix
+          path: vscode-bazel-*.vsix
+          if-no-files-found: error
+
+      - name: Upload Release Artifact
+        if: ${{ github.event_name == 'release' }}
+        shell: bash
+        run: |
+          upload_url=`echo '${{ github.event.release.upload_url }}' | cut -f1 -d"{"`;
```

Comment:
Why do we cut the upload URL at `{` here?

Reply:
One of those instances where I am unhappy with my former self for not commenting the code properly... I just copied this over from https://github.com/vogelsgesang/hyper-ir-lsp/blob/main/.github/workflows/build.yml, but don't remember anymore why I wrote it this way in the first place.

I remember this to be some "poor man's template parsing". I think the upload_url was something like `github.com/bazelbuild/vscode-bazel/releases/{some_placeholder}` and I wanted to strip the placeholder.

I will reverse-engineer this later tonight and add comments to the code

Reply:
Using the gh command line tool is maybe clearer here, like [this](https://github.com/cameron-martin/bazel-lsp/blob/98382215c44745485ed9ab7c1d13472a5acc0350/.github/workflows/build-release.yml#L87). You should be able to get the tag name as `github.event.release.tag_name`, and avoid any poor man's template parsing.

Reply:
Did you consider an existing composite Action to do this? For bazel rulesets, we use this one: https://github.com/bazel-contrib/.github/blob/master/.github/workflows/release_ruleset.yaml#L69-L77

(In general I think it would be nice for this automation to be similar to that of rulesets where possible)

Reply:
@alexeagle if understand correctly, the `action-gh-release@v1` creates a new release. What we are trying to achieve here is slightly different, though: We want to attach artifacts to an already existing release

Reply:
@cameron-martin good point. I updated the PR to use `gh release upload` instead